plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -125,6 +134,360 @@ protected TestTable createTableWithUnsupportedColumn()
+
+        // GROUP BY and WHERE on varchar column
+        // GROUP BY and WHERE on "other" (not aggregation key, not aggregation input)
+        assertConditionallyPushedDown(
+                getSession(),
+                "SELECT regionkey, sum(nationkey) FROM nation WHERE regionkey < 4 AND name > 'AAA' GROUP BY regionkey",
+                hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_INEQUALITY),
+                node(FilterNode.class, node(TableScanNode.class)));
+        // GROUP BY above WHERE and LIMIT
+        assertConditionallyPushedDown(
+                getSession(),
+                "SELECT regionkey, sum(nationkey) FROM (SELECT * FROM nation WHERE regionkey < 2 LIMIT 11) GROUP BY regionkey",
+                hasBehavior(SUPPORTS_LIMIT_PUSHDOWN),
+                node(LimitNode.class, anyTree(node(TableScanNode.class))));
+        // GROUP BY above TopN
+        assertConditionallyPushedDown(
+                getSession(),
+                "SELECT clerk, sum(totalprice) FROM (SELECT clerk, totalprice FROM orders ORDER BY orderdate ASC, totalprice ASC LIMIT 10) GROUP BY clerk",
+                hasBehavior(SUPPORTS_TOPN_PUSHDOWN),
+                node(TopNNode.class, anyTree(node(TableScanNode.class))));
+        // GROUP BY with JOIN
+        assertConditionallyPushedDown(
+                joinPushdownEnabled(getSession()),
+                "SELECT n.regionkey, sum(c.acctbal) acctbals FROM nation n LEFT JOIN customer c USING (nationkey) GROUP BY 1",
+                hasBehavior(SUPPORTS_JOIN_PUSHDOWN),
+                node(JoinNode.class, anyTree(node(TableScanNode.class)), anyTree(node(TableScanNode.class))));
+        // GROUP BY with WHERE on neither grouping nor aggregation column
+        assertConditionallyPushedDown(
+                getSession(),
+                "SELECT nationkey, min(regionkey) FROM nation WHERE name = 'ARGENTINA' GROUP BY nationkey",
+                hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_EQUALITY),
+                node(FilterNode.class, node(TableScanNode.class)));
+        // aggregation on varchar column
+        assertThat(query("SELECT min(name) FROM nation")).isFullyPushedDown();
+        // aggregation on varchar column with GROUPING
+        assertThat(query("SELECT nationkey, min(name) FROM nation GROUP BY nationkey")).isFullyPushedDown();
+        // aggregation on varchar column with WHERE
+        assertConditionallyPushedDown(
+                getSession(),
+                "SELECT min(name) FROM nation WHERE name = 'ARGENTINA'",
+                hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_EQUALITY),
+                node(FilterNode.class, node(TableScanNode.class)));
+
+        // TODO array_agg returns array, so it could be supported
+        assertThat(query("SELECT array_agg(nationkey) FROM nation")).isNotFullyPushedDown(AggregationNode.class);
```

Comment:
`array_agg` doesn't guarantee ordering of elements and hence is flaky depending on the system we test with. Any suggestions how to handle this?

Reply:
I extracted this and other unsupported ones to a new method so that they can be easily overridden or excluded where required.